test/linter.js

```diff
@@ -4,23 +4,124 @@
+    const vrx = /['"](.*?)['"]/g;
+    let v = vrx.exec(match[1]);
+    if (v && v.length && result.indexOf(v[1]) == -1) {
+      result.push(v[1]);
+    }
+    match = rx.exec(content);
+  }
+  return result;
+}
+
+async function getTagsFromConfig(config) {
+  // get hold of all tags and their corresponding input files using the literate config tool
+  const tagsMap = await getTagsMapFromConfig({ file: config });
+  if (tagsMap === null) {
+    return null;
+  }
+  const tags = Object.keys(tagsMap);
+
+  // filter the tags
+  if (utils.prOnly) {
+    // get path to the modified files relative to their corresponding md file, need to do this since 
+    // config files have relative paths to the input files
+    let allModifiedFiles = utils.getFilesChangedInPR();
+    allModifiedFiles = allModifiedFiles.map(mfile => {
+      return mfile.replace(path.dirname + '/', '');
+    });
+
+    // for each tag->files, find if there are any modified files and select those tags
+    return tags.filter(tag => {
+      const tagFiles = (tagsMap[tag] + '').split(',');
+      // find intersection with the modified files
+      return tagFiles.filter(tagFile => {
+        return allModifiedFiles.indexOf(tagFile) > -1;
+      }).length > 0;
+    });
+  }
+  return tags;
+}
+
+function execLinterCommand(config, tag) {
+  var cmd = `autorest --validation --azure-validator ${config} --message-format=json ${tag}`.trim();
+  console.log(`Executing: ${cmd}`);
+  try {
+    let result = execSync(cmd, { encoding: 'utf8', maxBuffer: 1024 * 1024 * 64 });
+  } catch (err) {
+    throw new Error('Linter validation contains error(s)');
+  }
+
+}
 
 describe('AutoRest Linter validation:', function () {
-  let configsToProcess = utils.getConfigFilesChangedInPR();
   // Useful when debugging a test for a particular swagger. 
   // Just update the regex. That will return an array of filtered items.
   // configsToProcess = ['specification/sql/resource-manager/readme.md'];
+  let configsToProcess = utils.getConfigFilesChangedInPR();
   for (const config of configsToProcess) {
     it(config + ' should honor linter validation rules.', async function () {
-      var cmd = `autorest --validation --azure-validator ${config} --message-format=json`;
-      console.log(`Executing: ${cmd}`);
 
-      try {
-        let result = execSync(cmd, { encoding: 'utf8', maxBuffer: 1024 * 1024 * 64 });
-      } catch (err) {
-        throw new Error('Linter validation contains error(s)');
+      // find all tags in the config file
+      const tagsToProcess = await getTagsFromConfig(config);
+      // if no tags found to process, run with the defaults    
+      if (tagsToProcess === null || tagsToProcess.length === 0) {
+        // no tags found
+        // this means we need to run validator against the individual 
+        // json files included in the PR
+        // but in the same directory tree as the config file
+        const filesChangedInPR = utils.getFilesChangedInPR();
```

Comment:
not sure about this - this case indicates a huge data quality problem in the repo. Sure, maybe do some validation even in that case, but maybe just throwing would be better so the PR that creates the issue is rejected!